src/viam/components/board/board.py

```diff
@@ -390,3 +394,25 @@ async def write_analog(self, pin: str, value: int, *, timeout: Optional[float] =
             value (int): value to write.
         """
         ...
+
+    @abc.abstractmethod
+    async def stream_ticks(
+        self, interrupts: List[DigitalInterrupt], *, timeout: Optional[float] = None, **kwargs
+    ) -> Stream[Tick]:
```

Comment:
(minor) since we defined `TickStream` above, should we not use it here in the return?